api.md

```diff
@@ -0,0 +1,73 @@
+* `/api`
+    * `GET` - Information about the API (version, metadata, links to repo, supported endpoints) - TODO
+
+* `/api/years`
+    * `GET` - List of all supported years. Structure: `{"years": ["2017"]}`
+
+* `/api/years/<year>`
+    * `GET` - Detailed attributes/metadata for a given year. Structure: [years](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/year.json)
+
+* `/api/years/<year>/institutions`
+    * `GET` - Complete list of institutions a user can/could file for a given year. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json)
+
+* `/api/institutions/`
+    * `GET` - Complete list of institutions a user can/could file for *across all years*. Structure: [institutions](https://github.com/cfpb/hmda-platform-ui/blob/c7590b1e6bd3243a8cd7909463ab8e2dc8c046a5/server/json/user1-institutions.json), separated by year.
```

Comment:
How do we get the user? This endpoint needs user information or else it would return the whole panel. I guess it's implied that user information will be delivered inside the request (headers?)